.github/workflows/sonarcloud-check.yml

```
@@ -0,0 +1,58 @@
+name: SonarCloud Check
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - source/**
+      - .github/workflows/sonarcloud.yml
```

```suggestion
      - .github/workflows/sonarcloud-check.yml
```

or rename the file and fix the other trigger.

---

I have only restored the previous configuration. That file contained `-check` and the configuration was updated to reflect this. 

What would you like the end result to be? I will change accordingly.

---

I think "check" informs us that the workflow doesn't "do" anything (like creating artifacts) but solely performs a check. So I'd go with "-check".